CRM/Admin/Form/PaymentProcessor.php

```diff
@@ -400,4 +401,34 @@ public function updatePaymentProcessor(&$values, $domainID, $test) {
     civicrm_api3('PaymentProcessor', 'create', $params);
   }
 
+  /**
+   * Check if there is a record with the same name in the db.
+   *
+   * @param string $value
+   *   The value of the field we are checking.
+   * @param array $options
+   *   The daoName and fieldName (optional ).
+   *
+   * @return bool
+   *   true if object exists
+   */
+  public static function paymentProcessorNameExists($value, $options) {
+    $fieldName = 'name';
+    $daoName = CRM_Utils_Array::value(0, $options);
```

Comment:
Its interesting to know that you already had a similar conversations .... tbh this Idea came to my mind two years ago when I was starting a new project ( by creating an API and consume it in the same application ) but I thought it's a stupid  idea since the general concept about API is they to be consumed  by other systems and apps not by the system itself for which they are belongs . Anyway thanks for pointing that out.